### PR TITLE
Add a GUI test to test the error dialog #133

### LINKED ISSUE #133: Add a GUI test to test the error dialog
e.g. We can trigger a DataSavingExceptionEvent and check if the GUI responds accordingly.

### PR DESCRIPTION
**Pull request by yl-coder**
Fixes #133

### CODE REVIEW

**Comment by yl-coder** [comment]
@damithc Have a look before i continue with the rest of the error dialogs.

File: src/test/java/guitests/AddressBookGuiTest.java [hunk 1]
```diff
@@ -114,4 +116,8 @@ protected void assertListSize(int size) {
     protected void assertResultMessage(String expected) {
         assertEquals(expected, resultDisplay.getText());
     }
+
+    public void raise(BaseEvent e) {
+        Platform.runLater(() -> EventsCenter.getInstance().post(e));
```
Comment (damithc) [inline src/test/java/guitests/AddressBookGuiTest.java]: Add a comment to explain this `Platform.runLater()` thing.

File: src/test/java/guitests/guihandles/AlertDialogHandle.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package guitests.guihandles;
+
+import guitests.GuiRobot;
+import javafx.scene.control.DialogPane;
+import javafx.stage.Stage;
+import seedu.address.ui.UiManager;
+
+/**
+ * A handler for the AlertDialog of the UI
```
Comment (damithc) [inline src/test/java/guitests/guihandles/AlertDialogHandle.java]: handle

File: src/test/java/guitests/guihandles/AlertDialogHandle.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+public class AlertDialogHandle extends GuiHandle {
+
+
+    public AlertDialogHandle(GuiRobot guiRobot, Stage primaryStage, String dialogTitle) {
+        super(guiRobot, primaryStage, dialogTitle);
+    }
+
+    public boolean isAlertDialogValid(String headerMessage, String contentMessage) {
```
Comment (damithc) [inline src/test/java/guitests/guihandles/AlertDialogHandle.java]: `isMatching( _ , _ )`?

File: src/test/java/guitests/guihandles/GuiHandle.java [hunk 4]
```diff
@@ -18,6 +19,7 @@
 public class GuiHandle {
     protected final GuiRobot guiRobot;
     protected final Stage primaryStage;
+    protected Optional<Stage> intermediateStage = Optional.empty();
```
Comment (damithc) [inline src/test/java/guitests/guihandles/GuiHandle.java]: javadoc comment?

**Comment by yl-coder** [comment]
Seems like this is the only alert dialog that is testable. Another appearance of the alert dialog is the error occured during initialisation the UI components, like FXML errors.

**Comment by damithc** [comment]
Yes, it will be hard to test errors that happen during app launch, especially those related to FXML.

**Pull request merged**